packages/flutter/lib/src/material/menu_anchor.dart

```diff
@@ -3156,7 +3156,12 @@ class _MenuLayout extends SingleChildLayoutDelegate {
       } else if (offBottom(y)) {
         final double newY = anchorRect.top - childSize.height;
         if (!offTop(newY)) {
-          y = newY;
+          // Only move the menu up if it is the root menu
```

Comment:
Period at the end of the sentence here and in the comments in the test below.

Reply:
Done